### DISCUSSION TITLE
Firefox 3rdparty Policy support

### DISCUSSION THREAD

**Discussion by lomenzel** [piroor/treestyletab#3552] [Q&A] [answered]
Hey there, 
does TST support configuration via the Firefox policies?

https://extensionworkshop.com/documentation/enterprise/enterprise-development/#how-to-add-policy

**Comment by piroor** [comment, marked as answer]
Yes, TST (and other my Firefox addons) supports configuring via GPO and policies.json.
Here is the complete list of configurable options:
https://github.com/piroor/treestyletab/blob/02abdd1d40aa2efee1eaebe7190cfa90c721fca4/webextensions/common/common.js#L129-L591

For example:

```json
{
  "policies": {
    "3rdparty": {
      "Extensions": {
        "treestyletab@piro.sakura.ne.jp": {
          "sidebarPosition": 1,
            "sidebarPosition:locked": false
        }
      }
    }
  }
}
```

I recommend you to configure TST by hand and export all options, then copy exported data and paste it under `policy.3rdparty.Extensions.treestyletab@piro.sakura.ne.jp`. They are locked by default but you can allow users to change the option with setting  `"<name>:locked":false` (so it means "changing the default value").
